From 7a1d131afbb1aee15b56d8dd2405f270bbc486c2 Mon Sep 17 00:00:00 2001
From: Runxi Yu <runxiyu@umich.edu>
Date: Wed, 11 Mar 2026 22:08:09 +0800
Subject: [PATCH] os: use t.TempDir in root_test tempDirWithUnixSocket

tempDirWithUnixSocket used os.MkdirTemp("", ""), while other paths in
root consistency tests use t.TempDir().

If GOTMPDIR and the system temp dir are on different filesystems (such
as during abuild), TestRootConsistencyRename and TestRootConsistencyLink
may produce EXDEV.

Fixes #78063

Change-Id: Ib81abec0876babf0cc0e2d9e53e34f051ddfbbc6
Reviewed-on: https://go-review.googlesource.com/c/go/+/754102
Auto-Submit: Sean Liao <sean@liao.dev>
Auto-Submit: Damien Neil <dneil@google.com>
Reviewed-by: David Chase <drchase@google.com>
Reviewed-by: Sean Liao <sean@liao.dev>
LUCI-TryBot-Result: Go LUCI <golang-scoped@luci-project-accounts.iam.gserviceaccount.com>
Reviewed-by: Damien Neil <dneil@google.com>
---
 src/os/root_test.go | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/src/os/root_test.go b/src/os/root_test.go
index 71ce5e559a2dba..2a6d20816f2281 100644
--- a/src/os/root_test.go
+++ b/src/os/root_test.go
@@ -1211,15 +1211,7 @@ var rootConsistencyTestCases = []rootConsistencyTest{{
 }}
 
 func tempDirWithUnixSocket(t *testing.T, name string) string {
-	dir, err := os.MkdirTemp("", "")
-	if err != nil {
-		t.Fatal(err)
-	}
-	t.Cleanup(func() {
-		if err := os.RemoveAll(dir); err != nil {
-			t.Error(err)
-		}
-	})
+	dir := t.TempDir()
 	addr, err := net.ResolveUnixAddr("unix", filepath.Join(dir, name))
 	if err != nil {
 		t.Skipf("net.ResolveUnixAddr: %v", err)
